core/iwasm/interpreter/wasm_runtime.c

```diff
@@ -3231,3 +3234,26 @@ llvm_jit_free_frame(WASMExecEnv *exec_env)
           || WASM_ENABLE_PERF_PROFILING != 0 */
 
 #endif /* end of WASM_ENABLE_JIT != 0 || WASM_ENABLE_WAMR_COMPILER != 0 */
+
+#if WASM_ENABLE_LIBC_WASI != 0 && WASM_ENABLE_MULTI_MODULE != 0
+void
+wasm_broadcast_wasi_args(WASMModule *module)
```

Comment:
How about changing to `wasm_propagate_wasi_args` or `wasm_transfer_wasi_args`? `broadcast` seems to be behaviors among different threads.